backend/src/controllers/dish.ts

```diff
@@ -1,94 +1,68 @@
 import { Request, Response } from 'express'
 
-import { db } from '../services/firebase'
 import { Dish } from '../models/dish'
 import { Transaction } from '../models/transaction'
-import { mapDishesToLatestTransaction, mapToDishVM } from '../services/dish'
+import {
+    getAllDishes,
+    getAllUserDishes,
+    getAllUserDishesInUse,
+    getAllUserDishesVM,
+    mapDishesToLatestTransaction,
+    mapToDishVM,
+} from '../services/dish'
 import { CustomRequest } from '../middlewares/auth'
+import { verifyIfUserAdmin } from '../services/users'
+import { getAllTransactions } from '../services/transactions'
 import Logger from '../utils/logger'
 
 export const getDishes = async (req: Request, res: Response) => {
-    // TODO: send dish information based on user role
+    let userClaims = (req as CustomRequest).firebase
+    let all = req.query['all']?.toString()
+    let withTransactions = req.query['transaction']?.toString()
+    let borrowed = req.query['borrowed']?.toString()
 
-    // Example of how to get user claims
-
-    // let userClaims = (req as CustomRequest).firebase
-    // userClaims.role
-
-    // get dishes from firebase
-    let rawDishData = <Array<Dish>>[]
+    let allDishes = <Array<Dish>>[]
     try {
-        let dishesQuerySnapshot = await db.collection('dishes').get()
-        dishesQuerySnapshot.docs.forEach((doc) => {
-            let data = doc.data()
-            rawDishData.push({
-                id: doc.id,
-                qid: parseInt(data.qid, 10),
-                registered: data.registered.toDate(),
-                type: data.type ? data.type : '',
-            })
+        allDishes = await getAllDishes()
+        Logger.info({
+            message: 'retrieved all dishes',
         })
-    } catch (e) {
+    } catch (err: any) {
         Logger.error({
-            error: e,
-            message: 'Error when fetching dishes from firebase',
-            statusCode: 500,
+            error: err.message,
+            statusCode: 500
         })
-        res.status(500).json({ error: 'internal_server_error' })
-        return
+        return res.status(500).json({ error: 'internal_server_error', message: err.message })
     }
 
-    Logger.info({
-        message: 'Retrieved dish data from firebase',
-    })
+    if (withTransactions !== 'true' && all == 'true') {
```

Comment:
Can you do a strict check?
```suggestion
    if (withTransactions !== 'true' && all === 'true') {
```